SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -552,7 +552,34 @@ - (BOOL)sdl_isDynamicMenuUpdateActive:(SDLDynamicMenuUpdatesMode)dynamicMenuUpda
 }
 
 - (BOOL)sdl_artworkNeedsUpload:(SDLArtwork *)artwork {
-    return (artwork != nil && ![self.fileManager hasUploadedFile:artwork] && !artwork.isStaticIcon);
+    if (artwork != nil) {
+        if (artwork.isStaticIcon) {
+            return NO;
+        } else {
+            return (artwork.overwrite || (self.fileManager != nil && ![self.fileManager hasUploadedFile:artwork]));
+        }
+    }
+    return NO;
+}
+
+- (BOOL) sdl_allArtworksUploaded:(NSArray<SDLMenuCell *> *)cells {
+    for (SDLMenuCell *cell in cells) {
+        if (![self sdl_artworkUploaded:cell.icon]) {
+            return NO;
+        }
+        if (cell.subCells != nil && (cell.subCells.count) > 0) {
+            return [self sdl_allArtworksUploaded:cell.subCells];
+        }
+    }
+
+    return YES;
+}
+
+- (BOOL) sdl_artworkUploaded:(SDLArtwork *)artwork {
```

Comment:
```suggestion
- (BOOL)sdl_isArtworkUploaded:(nullable SDLArtwork *)artwork {
```

Remember that everything is assumed to be `nonnull` unless you explicitly specify it as `nullable`.